packages/protocol/lib/compatibility/ast-layout.ts

```diff
@@ -131,39 +158,51 @@ const compareStructDefinitions = (oldType: TypeInfo, newType: TypeInfo) => {
   }
 }
 
-const generateStructsCompatibilityReport = (oldLayout: StorageLayoutInfo, newLayout: StorageLayoutInfo) => {
+// Struct is expandable only if used in mappings or arrays
+const isStructExpandable = (oldType: TypeInfo, oldLayout: StorageLayoutInfo) => {
+  const structString = `t_struct<${oldType.label}>`
+  return !oldLayout.storage.some(storage => storage.type === structString)
```

Comment:
For completeness, this should also check there are no fixed-length arrays (as top level storage variables, or nested inside other complex types).